### ISSUE TITLE
Description field erroneously reported as missing from dataset_description & unexpected warnings for derivatives

### ISSUE THREAD

**Issue by LuciMoore** [bids-standard/bids-validator#202]
I’m trying to resolve some of the warnings listed for our [BOBS Repo dataset on OpenNeuro](https://openneuro.org/datasets/ds005450/versions/1.0.0), which are derivatives containing a T1w, T2w, and manually corrected brain segmentation per subject/session

One issue is that it says the [dataset_description.json](https://openneuro.org/datasets/ds005450/versions/1.0.0/file-display/dataset_description.json) is missing the field Description even though it is not. I've tried several fixes and get the same warnings both when running `deno run -A jsr:@bids/validator` locally and using the [online BIDs validator](https://bids-standard.github.io/bids-validator/). **I also get this warning when I copy the contents of the example provided in the BIDS documentation [here](https://bids-specification.readthedocs.io/en/stable/modality-agnostic-files.html#derived-dataset-and-pipeline-description).** 

The other main issue I'm having is that there are a very large number of warnings telling me to add fields relevant to raw BIDS data acquisition, not derivatives (eg `ReceiveCoilName`), to the jsons. The warnings go away when I add these fields, but they are fields that shouldn't be included in derivative metadata to begin with. I'm only including sidecar jsons at all because the files are in a non-standard space (MNI infant), so I need to specify `SpatialReference` in a sidecar json

I know these are just warnings and not errors, but if they are possible to resolve that would be great as it is being requested by a reviewer for our manuscript about this open resource (and also would be great to follow BIDS spec as closely as possible).

**Comment by effigies** [comment]
Hi @LuciMoore, I ran the validator on the dataset and here are some notes:

1) The dataset_description issue isn't a `Description` field, but `SourceDatasets`:

```
	[WARNING] JSON_KEY_RECOMMENDED A JSON file is missing a key listed as recommended.
		SourceDatasets
		/dataset_description.json - Field description: Used to specify the locations and relevant attributes of all source datasets.
Valid keys in each object include `"URL"`, `"DOI"` (see
[URI](SPEC_ROOT/common-principles.md#uniform-resource-indicator)), and
`"Version"` with
[string](https://www.w3schools.com/js/js_json_datatypes.asp)
values.


	Please visit https://neurostars.org/search?q=JSON_KEY_RECOMMENDED for existing conversations about this issue.
```

If there are no citable source datasets, then setting `"SourceDatasets": []` will suppress the warning.

2) A lot of these other issues are related to https://github.com/bids-standard/bids-specification/issues/2040, which is basically a problem where BIDS dumped a wishlist of nice-to-have metadata and called in RECOMMENDED. Now that the validator takes RECOMMENDED seriously, I think we should pare this down, but I got some pushback that led to that stalling. I will

I would note that warnings are not errors because they may not apply to every dataset. Metadata that is not available can be worth notifying a curator about, but that does not mean that the metadata can be generated. We should try to minimize them, but ultimately it is up to the dataset curator to review the warnings and determine that they do not apply. I think 0 warnings is an unreasonable demand of a reviewer.

I can't find the issue, but @yarikoptic has proposed allowing something like a `.bids-validator-config.json` to allow warnings to be explicitly silenced. That would allow you to basically mark issues as "known but nothing to be done".

cc @nellh The rendering of warnings on this dataset looks sparse. Can we rerun it to utilize updated fields for rendering?

**Comment by LuciMoore** [comment]
@effigies thanks for the quick response! I was troubleshooting locally and did add `SourceDatasets`, but still get the same warning:
```
	[WARNING] SIDECAR_KEY_RECOMMENDED A data file's JSON sidecar is missing a key listed as recommended.
		Description
		/README.md - Field description: Free-form natural language description.
```

 I've attached the updated `dataset_description.json` for reference. I'm really unclear on what is happening here.

[dataset_description.json](https://github.com/user-attachments/files/20106764/dataset_description.json)

The information provided about the 2nd point I raised is sufficient for me to close that off, thanks!

**Comment by LuciMoore** [comment]
I very carefully updated the `dataset_description.json` again to follow the [documentation](https://bids-specification.readthedocs.io/en/stable/modality-agnostic-files.html#dataset_descriptionjson) and am still getting this error 

[dataset_description.json](https://github.com/user-attachments/files/20109158/dataset_description.json)

**Comment by LuciMoore** [comment]
@effigies Final thoughts - I'm fine with just ignoring the "Description" warnings for the several filetypes it pops up for. It's just so curious to me that I get this warning whether the field is included or not, and I also don't see any mention of this key in the documentation either - eg for [dataset_description](https://bids-specification.readthedocs.io/en/stable/modality-agnostic-files.html#dataset_descriptionjson), I ended up just leaving it out as a top-level key because I only see it mentioned as optional for `GeneratedBy` array. Let me know if you have thoughts, otherwise you can go ahead and close this issue! Thanks for your help!

From OpenNeuro BIDS validation outputs for quick reference:
```
SIDECAR_KEY_RECOMMENDED - Description
Rule: rules.sidecars.derivatives.common_derivatives.CommonDerivativeFields
Messages:
A data file's JSON sidecar is missing a key listed as recommended.
```

**Comment by effigies** [comment]
Okay, figured out the `Description` thing: https://github.com/bids-standard/bids-validator/issues/202

The idea for that rule was that any file that could have a sidecar should have a Description field. The problem is there isn't a selector for which files can have sidecars. There are no such broad rules in raw datasets.

**Comment by LuciMoore** [comment]
@effigies I'm probably stating the obvious, but I assume the bids validator in OpenNeuro will be updated with these changes at some point, correct?

**Comment by effigies** [comment]
@rwblair We need a release before that.

**Comment by effigies** [comment]
Released. Pending OpenNeuro deployment.
